components/vocabularies/api/src/main/java/org/phenotips/vocabulary/internal/solr/AbstractOBOSolrVocabulary.java

```diff
@@ -118,12 +128,23 @@ protected int index(String sourceUrl)
                     termBatch = new HashSet<>();
                     batchCounter = 0;
                 }
+                singleValuedFileds.forEach(fieldName -> singleValueFieldsTracker.put(fieldName, false));
```

Comment:
you only need `singleValueFieldsTracker` here, so the right way is to define it here, and initialize it here.

You also don't need a `Map` in that case, it can be done much simpler, as a `Set`, where a presense of  afield indicates it has been encountered already, no need to add a `true` to that

E.g. on this line initialize `Set<String> alreadyPresentFields = new HashSet<>();`

then add all fields encountered on line #146 (can even remove an `if` to reduce cyclomatic complexity).

this way also never have to clear the Set